### ISSUE TITLE
On prem Bitbucket integration support.

### ISSUE THREAD

**Issue by ColdCerealAtWork** [a85/POSTMan-Chrome-Extension#8769]
I can't take advantage of Bitbucket integration because my company uses Bitbucket Server instead of Cloud.

I'd love to be able to integrate with an on prem bitbucket server.

**Comment by ogtool** [comment]
Agree. We have teams wanting to connect Postman with our on-prem Bitbucket environment. Instead of making a Bitbucket on-prem specific integration, is it worth just supporting generic git (hostname/user/pass|ssh) ?

**Comment by fclante** [comment]
This Is critical functionality Please prioritize it SOON. My Company wants to use the product, but it's not possible if we have to wait forever for this much needed feature.

**Comment by abhijitkane** [comment]
@KristianAtVsp @ogtool @fclante Can you describe the workflow that you'd like to achieve with this integration? Are your Bitbucket instances accessible from the internet? We recently added support for Static IPs for Integrations (https://blog.postman.com/how-to-integrate-postman-with-your-private-network/), which means there's a single IP that you can allow to call your network.

**Comment by fclante** [comment]
Hi @abhijitkane We have fairly ordinary setup. We design api's in Postman, and we want to commit the along with our source code in our Bitbucket repositories (Bitbucket Server) . We don't expose our Bitbucket instance on the public internet and would prefer not to allow static ip-access through our firewall (although it's possible). We are about 300+ developers spread over several departments and teams. I'm tasked with a POC for using Postman and a part of that is proving that we can benefit by keeping our collection-tests alongside our source-code in bitbucket. We would really appreciate a Bitbucket-Server integration feature from Postman ASAP. Merry Christmas - Frey Clante.

**Comment by fclante** [comment]
@jotsarup Hi I would like to hear how far you are with implementing this much anticipated feature?
It's of vital importance if we are to keep on using postman as a tool in our organization that we can have bitbucket server integration.

**Comment by ChrisRo89** [comment]
Would be pretty cool to have this integration, I'm wondering how big the effort for it is - as the Rest API looks similar on-premise then for the cloud.

**Comment by futtie** [comment]
My company is currently working on an integration, where an on-prem Bitbucket solution is mandatory - all our code is on that setup, and it is not accessible from the outside. Quite similar to the other setups mentioned here. 
What's the status of this request? I can see that it's in "Pending Triage", but what does that mean - days, weeks, months, years?

**Comment by landofbunnies** [comment]
Would also very much like to see this in a future release. We currently cannot utilize Postman as well as we would like due to our on-premises implementation of Bitbucket.

**Comment by fclante** [comment]
Atlassian has decided to end support for server products : https://www.atlassian.com/migration/assess/journey-to-cloud

**Comment by ChrisRo89** [comment]
@fclante Atlassian still Sells there Data Center Products as alternative: "Support for Atlassian Server products ends on February 2, 2024, but you can stay supported and benefit from continued innovation by migrating to our Cloud or Data Center products." [https://www.atlassian.com/migration/assess/journey-to-cloud] which means there are still Bitbucket On-premise and this options is still interesting

**Comment by MatusDallos** [comment]
Hello @abhijitkane @jotsarup, 
please, is there any timeline or progress for this feature? 
Our company would like to use this integration for Atlassian Data Center products as well. 
Thank you!

**Comment by shashankawasthi88** [comment]
MatusDallos and All,

The team is actively working on the integration, we will update the thread once we have final timelines decided.

Thanks

**Comment by emilytrabert** [comment]
It's been 7 months since the last post - any updates from the team?

**Comment by git-ew** [comment]
Also interested in this integration.

**Comment by fclante** [comment]
As for our part this is no longer a need feature because we've migrated to Github.

**Comment by lanes100** [comment]
I can't believe this still hasn't been worked on after almost 4 years. We are using Bitbucket Datacenter and can't integrate Postman with it still...

**Comment by shashankawasthi88** [comment]
Hi All ,

As of [Postman v12](https://www.postman.com/downloads/), we now support Native Git Integration.
You can connect a workspace directly to a local Git repository and work against your filesystem. Collections and related artifacts are stored as files, enabling standard Git workflows (branching, diffs, PRs, CI/CD) without manual export/import. This will allow you to commit and push changes to your on-prem BitBucket repositories.
Docs: https://learning.postman.com/docs/agent-mode/native-git
Announcement: https://blog.postman.com/new-postman-is-here/
Feel free to try it out and share feedback.

https://github.com/user-attachments/assets/856132c1-676c-4d31-915f-6e5519919dee
